test/runtests.jl

```diff
@@ -71,3 +79,40 @@ end
     @test CustomSetproperties(2) === setproperties(o, a=2)
     @test CustomSetproperties(2) === setproperties(o, (a=2,))
 end
+
+@testset "Default object constructors" begin
+    A1 = zeros(5, 6)
+    A2 = ones(Float32, 5, 6)
+    @testset "SubArray" begin
+        subarray = view(A1, 1:2, 3:4)
+        fields = map(fn -> getfield(subarray, fn), fieldnames(typeof(subarray)))
```

Comment:
Should this be `getproperties` instead?

Reply:
yep

Reply:
Sorry I forgot to revisit the tests after these changes. I'll use `getproperties`